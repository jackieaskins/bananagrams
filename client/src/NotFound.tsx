import React from 'react';

const NotFound: React.FC = () => <h1>Not found</h1>;

export default NotFound;
