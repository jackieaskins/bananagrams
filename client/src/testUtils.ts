import {
  RenderOptions,
  RenderResult,
  render as rtlRender,
} from "@testing-library/react";
import userEvent, { UserEvent } from "@testing-library/user-event";

// https://testing-library.com/docs/user-event/intro
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult & { user: UserEvent } {
  return {
    user: userEvent.setup(),
    ...rtlRender(ui, options),
  };
}
