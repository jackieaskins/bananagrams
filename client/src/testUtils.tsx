import React from 'react';
import { shallow } from 'enzyme';

type CallbackFn = (...args: any[]) => any;

const TestHook = ({ callback }: { callback: CallbackFn }): null => {
  callback();
  return null;
};

export const testHook = (callback: CallbackFn): void => {
  shallow(<TestHook callback={callback} />);
};
