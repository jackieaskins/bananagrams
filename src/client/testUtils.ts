import {
  RenderOptions,
  RenderResult,
  render as rtlRender,
} from "@testing-library/react";
import userEvent, { UserEvent } from "@testing-library/user-event";

export const CURRENT_PLAYER_ID = "currentPlayerId";

// https://testing-library.com/docs/user-event/intro
export function renderComponent(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult & { user: UserEvent } {
  return {
    user: userEvent.setup(),
    ...rtlRender(ui, options),
  };
}
