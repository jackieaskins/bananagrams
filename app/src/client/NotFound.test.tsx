import NotFound from "./NotFound";
import { renderComponent } from "./testUtils";

describe("<NotFound />", () => {
  it("renders not found header", () => {
    const { asFragment } = renderComponent(<NotFound />);

    expect(asFragment()).toMatchSnapshot();
  });
});
