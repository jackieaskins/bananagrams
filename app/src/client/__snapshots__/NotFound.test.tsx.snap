// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders not found header 1`] = `
<DocumentFragment>
  <div
    class="css-h74a88"
  >
    <div
      class="chakra-container css-0"
    >
      <h1
        class="chakra-heading css-xi606m"
      >
        404: Page Not Found
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
